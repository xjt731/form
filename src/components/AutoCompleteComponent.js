import { Button, DatePicker, Form, Input, AutoComplete } from 'antd';
import { useState } from 'react';

export default function AutoCompleteComponent(){
    // text hints used for AutoComplete
  const [options, setOptions] = useState([]);
  //return a object, and it which could be set how many text hints are able to appear on screen 
  const mockVal = (str, repeat = 1) => ({
    value: str.repeat(repeat),
  });
  //return an array which could be changed 
  const getPanelValue = (searchText) =>
    !searchText ? [] : [mockVal(searchText), mockVal(searchText, 2), mockVal(searchText, 3)];
  //collecting all information that users type in this form
  const onFinish = (values) => {
    console.log(values)
  }
    return (
        <AutoComplete
              options={options}
              style={{
                width: 200,
              }}
              onSearch={(text) => setOptions(getPanelValue(text))}
              placeholder="input here"
            />
    )
}
