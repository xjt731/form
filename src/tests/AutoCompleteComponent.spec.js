import { render, screen, fireEvent } from "@testing-library/react";
import AutoCompleteComponent from "../components/AutoCompleteComponent";

describe("AutoCompleteComponent", () => {
    test("render", () => {
        const { container } = render(<AutoCompleteComponent />)
        expect(container).toMatchSnapshot()
        const element = screen.getAllByRole("combobox");
        expect(element).toBeDefined()
    })



    test("user input", () => {
        render(<AutoCompleteComponent />);
        const inputElement = screen.getByRole("combobox");
        fireEvent.change(inputElement, { target: { value: "abc" } });
        expect(inputElement.value).toStrictEqual('abc')
    });

    

});

