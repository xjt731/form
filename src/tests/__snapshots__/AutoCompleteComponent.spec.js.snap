// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoCompleteComponent render 1`] = `
<div>
  <div
    class="ant-select ant-select-auto-complete css-dev-only-do-not-override-1me4733 ant-select-single ant-select-show-search"
    style="width: 200px;"
  >
    <div
      class="ant-select-selector"
    >
      <span
        class="ant-select-selection-search"
      >
        <input
          aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
          aria-autocomplete="list"
          aria-controls="rc_select_TEST_OR_SSR_list"
          aria-haspopup="listbox"
          aria-owns="rc_select_TEST_OR_SSR_list"
          autocomplete="off"
          class="ant-select-selection-search-input"
          id="rc_select_TEST_OR_SSR"
          role="combobox"
          type="search"
          value=""
        />
      </span>
      <span
        class="ant-select-selection-placeholder"
      >
        input here
      </span>
    </div>
  </div>
</div>
`;
